syntax = "proto3";
import "basic.proto";
import "common.proto";
import "parsing_base.proto";
package autodrive.workflow.proto;

enum ParkingSlotPointType {
  kParkingSlotPointTypeTruncated = 0;
  kParkingSlotPointTypeCorner = 1;
}

enum ParkingSlotType {
  kParkingSlotTypeUnknown = 0;
  kParkingSlotTypeVertical = 1;
  kParkingSlotTypeParallel = 2;
  kParkingSlotTypeOblique = 3;
}

message ParkingSlotPoint {
  Vector2f point = 1;
  float score = 2;
  ParkingSlotPointType point_type = 3;
  bool visibility = 4;
  float point_orientation_x = 5;
  float point_orientation_y = 6;
}

message ParkingSlotRaw {
  repeated ParkingSlotPoint slot_points = 1;
  float score = 2;
  bool occupancy = 3;
  ParkingSlotType slot_type = 4;
  float slot_orientation_x = 5;
  float slot_orientation_y = 6;
  int32 corner_match_flag = 7;
}

message IpmWheelGroundPoint {
  Vector2f point = 1;
  float score =2;
}

message IpmMultitaskMsg {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  repeated ParkingSlotRaw slots_raw = 3;
  repeated IpmWheelGroundPoint ipm_wheel_ground_points = 4;
  float test = 5;
}
